api/routes/books.js

```diff
@@ -26,53 +29,131 @@ router.get('/:id', async (req, res, next) => {
 
 // You should only be able to create a book if the user is an admin
 router.post('/', async (req, res, next) => {
-  const status = 200
+  const status = 201
   try {
+    //  Valid JWT is not provided 401
+    const token = req.headers.authorization.split('Bearer ')[1]
+    if (!token) {
+        const error = new Error (`Invalid token.`)
+        error.status = 401
+        next(error)
+    }
+    const payload = jsonwebtoken.verify(token, SECRET_KEY)
+    
+    // JWT token is for a user who is not an admin 401
+    const clientUser = await User.findOne({ _id: payload.id })
+      .select('-__v -password')
+    if (!clientUser.admin === true) {
+        const error = new Error (`Insufficient privelages.`)
+        error.status = 401
+        next(error)
+    }
+    
     const book = await Book.create(req.body)
     if (!book) throw new Error(`Request body failed: ${JSON.stringify(req.body)}`)
     
     const response = await Book.findById(book._id).select('-__v')
     res.json({ status, response })
   } catch (e) {
     console.error(e)
-    const message = 'Failure to create. Please check request body and try again.'
-    const error = new Error(message)
-    error.status = 400
-    next(error)
+    next(e)
   }
 })
 
 // You should only be able to reserve a book if a user is logged in
 router.patch('/:id/reserve', async (req, res, next) => {
   const { id } = req.params
   try {
+    const token = req.headers.authorization.split('Bearer ')[1]    
+    // A valid JWT token is not provided (status 401)
+    if (!token) {
```

Comment:
You don't want to just check whether or not there is a token, you want to check whether or not the token is valid. In this case, you should use `payload` from below.